src/coreclr/interpreter/compiler.cpp

```diff
@@ -1411,6 +1611,126 @@ bool InterpCompiler::CreateBasicBlocks(CORINFO_METHOD_INFO* methodInfo)
+                (codeStart + clause.TryOffset + clause.TryLength) > codeEnd)
+        {
+            return false;
+        }
+
+        InterpBasicBlock* pTryBB = GetBB(clause.TryOffset);
+
+        if ((codeStart + clause.HandlerOffset) > codeEnd ||
+                (codeStart + clause.HandlerOffset + clause.HandlerLength) > codeEnd)
+        {
+            return false;
+        }
+
+        // Find and mark all basic blocks that are part of the try region.
+        for (uint32_t j = clause.TryOffset; j < (clause.TryOffset + clause.TryLength); j++)
+        {
+            InterpBasicBlock* pBB = m_ppOffsetToBB[j];
+            if (pBB != NULL && pBB->clauseType == BBClauseNone)
+            {
+                pBB->clauseType = BBClauseTry;
+            }
+        }
+
+        InterpBasicBlock* pHandlerBB = GetBB(clause.HandlerOffset);
+
+        // Find and mark all basic blocks that are part of the handler region.
+        for (uint32_t j = clause.HandlerOffset; j < (clause.HandlerOffset + clause.HandlerLength); j++)
+        {
+            InterpBasicBlock* pBB = m_ppOffsetToBB[j];
+            if (pBB != NULL && pBB->clauseType == BBClauseNone)
+            {
+                if ((clause.Flags == CORINFO_EH_CLAUSE_NONE) || (clause.Flags == CORINFO_EH_CLAUSE_FILTER))
+                {
+                    pBB->clauseType = BBClauseCatch;
+                }
+                else
+                {
+                    assert((clause.Flags == CORINFO_EH_CLAUSE_FINALLY) || (clause.Flags == CORINFO_EH_CLAUSE_FAULT));
+                    pBB->clauseType = BBClauseFinally;
+                }
+            }
+        }
+
+        if (clause.Flags == CORINFO_EH_CLAUSE_FILTER)
+        {
+            if ((codeStart + clause.FilterOffset) > codeEnd)
+                return false;
+
+            // The filter funclet is always stored right before its handler funclet.
+            // So the filter end offset is equal to the start offset of the handler funclet.
+            InterpBasicBlock* pFilterBB = GetBB(clause.FilterOffset);
+            pFilterBB->isFilterOrCatchFuncletEntry = true;
+            pFilterBB->clauseVarIndex = m_clauseVarsIndex + i;
+
+            // Initialize the filter stack state. It initially contains the exception object.
+            pFilterBB->stackHeight = 1;
+            pFilterBB->pStackState = (StackInfo*)AllocMemPool(sizeof (StackInfo));
+            pFilterBB->pStackState[0].type = StackTypeO;
+            pFilterBB->pStackState[0].size = INTERP_STACK_SLOT_SIZE;
+            pFilterBB->pStackState[0].clsHnd = NULL;
+            pFilterBB->pStackState[0].var = pFilterBB->clauseVarIndex;
+
+            // Find and mark all basic blocks that are part of the filter region.
+            for (uint32_t j = clause.FilterOffset; j < clause.HandlerOffset; j++)
+            {
+                InterpBasicBlock* pBB = m_ppOffsetToBB[j];
+                if (pBB != NULL && pBB->clauseType == BBClauseNone)
```

Comment:
Will the clause type here be none if we set it to Catch for filters earlier?

Reply:
This is the filter funclet part of the filter. There cannot be any other funclet inside of a filter by ECMA. So theoretically, this second part of the condition could be turned into an assert in this specific case, but I'd slightly prefer keeping it the same as at other places.